pkg/plugins/runtime/gateway/register/enabled.go

```diff
@@ -5,18 +5,8 @@ import (
 	"github.com/kumahq/kuma/pkg/core/resources/registry"
 )
 
-// NOTE: this is non-deterministic in testing. Some tests will import
-// the plugin and trigger registration and some won't. This means that
-// whether the Gateway types are registered in tests depends on which
-// subset of tests are running.
 func RegisterGatewayTypes() {
-	// A Gateway is local to a zone, which means that it propagates in one
-	// direction, from a zone CP up to a global CP. The reason for this
-	// is that the Kubernetes Gateway API is the native Kubernetes API
-	// for Kuma gateways. If we propagated a Universal Gateway resource
-	// to a Kubernetes zone, we would need to be able to transform Gateway
-	// resources from Universal -> Kubernetes and have to deal with namespace
-	// semantics and a lot of other unpleasantness.
+	// RegisterTypeIfAbsent is used because it's not deterministic in testing that RegisterGatewayTypes is called only once.
```

Comment:
Do we know why it's not deterministic? Could/should it be?

Reply:
For example `BuildRuntime()` is a problem. It registers the plugin for every test case, but Type Registry is unfortunately global, so it tries to register multiple times.